types/src/account_config/events/new_block.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{account_address::AccountAddress, account_config::CORE_CODE_ADDRESS, event::EventKey};
+use anyhow::Result;
+use move_core_types::move_resource::MoveResource;
+use serde::{Deserialize, Serialize};
+
+/// Struct that represents a NewBlockEvent.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NewBlockEvent {
+    round: u64,
+    proposer: AccountAddress,
+    previous_block_votes: Vec<AccountAddress>,
+    time_micro_seconds: u64,
+}
+
+impl NewBlockEvent {
+    pub fn round(&self) -> u64 {
+        self.round
+    }
+
+    pub fn proposer(&self) -> AccountAddress {
+        self.proposer
+    }
+
+    pub fn previous_block_votes(&self) -> &[AccountAddress] {
+        &self.previous_block_votes
+    }
+
+    pub fn proposed_time(&self) -> u64 {
+        self.time_micro_seconds
+    }
+
+    pub fn try_from_bytes(bytes: &[u8]) -> Result<Self> {
+        lcs::from_bytes(bytes).map_err(Into::into)
+    }
+
+    pub fn event_key() -> EventKey {
+        EventKey::new_from_address(&CORE_CODE_ADDRESS, 14)
```

Comment:
I'm seeing duplicate declarations of this: https://github.com/libra/libra/blob/master/types/src/block_metadata.rs#L75-L77 can we try to clean this up, and maybe other event keys? maybe we need a better way to organize events and their keys too, long term